File: examples/fib.dart
int fibIterative(int n) {
  if (n <= 1) return n;
  var a = 0;
  var b = 1;
  for (var i = 2; i <= n; i++) {
    final t = a + b;
    a = b;
    b = t;
  }
  return b;
}

void main(List<String> args) {
  final iters = 50000;
  final n = 30;

  var acc = 0;
  for (var i = 0; i < iters; i++) {
    acc = fibIterative(n);
  }
  print('fib($n) = $acc, iters=$iters');
}

File: examples/_references/fib.lua
local function fib_iterative(n)
  if n <= 1 then return n end
  local a, b = 0, 1
  for _ = 2, n do
    local t = a + b
    a = b
    b = t
  end
  return b
end

local iters = 50000
local n = 30

local acc = 0
for _ = 1, iters do
  acc = fib_iterative(n)
end

print(string.format("fib(%d) = %d, iters=%d", n, acc, iters))

